mongo/integration/collection_test.go

```diff
@@ -883,6 +883,74 @@ func TestCollection(t *testing.T) {
 			_, ok := err.(mongo.CommandError)
 			assert.True(mt, ok, "expected error type %v, got %v", mongo.CommandError{}, err)
 		})
+		mt.Run("limit and batch size and skip", func(mt *mtest.T) {
+			var docs []interface{}
+			for i := 1; i <= 201; i++ {
+				docs = append(docs, bson.D{{"x", int32(i)}})
+			}
+
+			_, err := mt.Coll.InsertMany(mtest.Background, docs)
+			assert.Nil(mt, err, "InsertMany error for initial data: %v", err)
+
+			testCases := []struct {
+				limit     int64
+				batchSize int32
+				skip      int64
+			}{
+				{
+					99, 100, 10,
+				},
+				{
+					100, 100, 20,
+				},
+				{
+					80, 20, 90,
+				},
+				{
+					201, 201, 0,
+				},
+				{
+					100, 200, 120,
+				},
+			}
+
+			for _, tc := range testCases {
```

Comment:
I suggest adding a name to the test cases and having the loop use `mt.Run`:

```
mt.Run(tc.name, func(mt *mtest.T) { ... }
```

It will make it easier to identify which test case failed if they are split into cases. Even names as simple as "case 1", "case 2", "case 3", etc. will produce separate output for each test case.

Reply:
Ah good idea! Used "case 1", etc. since they were difficult to describe succinctly.